### PR TITLE
Fix how pre-generated keyshares are added for tests

### PR DESCRIPTION
**Pull request by ameba23**
https://github.com/entropyxyz/entropy-core/pull/1053 introduced random selection of initial signing set on jumpstart, but our pre-generated keyshares were always being added to alice, bob and charlies kvdbs, regardless of whether they were selected as initial signers. 

This fixes that - the pre-generated keyshares are given to whoever is initially selected to be a signer.

### CODE REVIEW

File: crates/testing-utils/src/helpers.rs [hunk 1]
```diff
@@ -25,11 +25,9 @@ use subxt::{backend::legacy::LegacyRpcMethods, OnlineClient};
 /// A helper for setting up tests which starts both a set of TS servers and a chain node and returns
 /// the chain API as well as IP addresses and PartyId of the started validators
 pub async fn spawn_tss_nodes_and_start_chain(
-    add_parent_key: bool,
     chain_spec_type: ChainSpecType,
 ) -> (OnlineClient<EntropyConfig>, LegacyRpcMethods<EntropyConfig>, Vec<String>, Vec<PartyId>) {
-    let (validator_ips, validator_ids) =
-        spawn_testing_validators(add_parent_key, chain_spec_type.clone()).await;
+    let (validator_ips, validator_ids) = spawn_testing_validators(chain_spec_type.clone()).await;
```
[the remark this answers is not in the document]
Comment (ameba23) [reply crates/testing-utils/src/helpers.rs:30]: Ty

File: crates/threshold-signature-server/src/helpers/tests.rs [hunk 2]
```diff
@@ -215,25 +210,38 @@ pub async fn spawn_testing_validators(
...
+
+    let port = match validator_name {
+        ValidatorName::Alice => 3001,
+        ValidatorName::Bob => 3002,
+        ValidatorName::Charlie => 3003,
+        ValidatorName::Dave => 3004,
+        _ => panic!("Unexpected validator name"),
+    };
+
+    // let port = 3001 + (validator_name as usize);
```
[the remark this answers is not in the document]
Comment (ameba23) [reply crates/threshold-signature-server/src/helpers/tests.rs:236]: Gonna do this :+1:

File: crates/threshold-signature-server/src/helpers/tests.rs [hunk 3]
```diff
@@ -284,16 +292,22 @@ pub async fn jump_start_network_with_signer(
...
         // Ignore the error as one confirmation will fail
-        let _result =
-            submit_transaction(api, rpc, &tss_signer, &jump_start_confirm_request, None).await;
+        if submit_transaction(api, rpc, &tss_signer, &jump_start_confirm_request, None)
+            .await
+            .is_ok()
+        {
+            put_keyshares_in_db(keyshare_index, validator_name).await;
+            keyshare_index += 1;
+        }
```
[the remark this answers is not in the document]
Comment (ameba23) [reply crates/threshold-signature-server/src/helpers/tests.rs:302]: The problem is the `await`.  Do you know how to do that async?

**Comment by ameba23** [comment]
Converting to draft as two tests are failing locally (And CI is stuck: https://github.com/entropyxyz/devops-infrastructure/issues/66 )

File: crates/threshold-signature-server/src/helpers/tests.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (dvdplm) [reply crates/threshold-signature-server/src/helpers/tests.rs:302]: I'm terrible at async rust but it feels like it should be possible. :/

**Comment by ameba23** [comment]
I am a little bit stuck on getting the reshare test to pass.

There are some issues with the block number having 1 subtracted from it which were not made clear before, because the OcwMessage which actually got passed from the chain was not needed since alice was no longer a signer, and the other OcwMessages which were mocked worked correctly.  Now we do need alice to participate in the reshare, and things are not working.

File: crates/threshold-signature-server/src/helpers/tests.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (JesseAbram) [reply crates/threshold-signature-server/src/helpers/tests.rs:302]: it should be ok to do a .and_then(|| async {  should be maybe probably

**Review by JesseAbram — APPROVED** [review]

**Pull request merged**